src/Nethermind/Nethermind.Synchronization/SnapSync/SnapProviderHelper.cs

```diff
@@ -56,6 +57,7 @@ public static (AddRangeResult result, bool moreChildrenToRight, List<PathWithAcc
             List<ValueHash256> codeHashes = new();
             bool hasExtraStorage = false;
 
+            using ArrayPoolList<PatriciaTree.BulkSetEntry> entries = new ArrayPoolList<PatriciaTree.BulkSetEntry>(accounts.Count);
```

Comment:
Should below loop be parallelized?

Reply:
Snap already parallelized at top level, should be fine to skip it here for efficiency.